File: backend/app/schemas/analytics.py
from pydantic import BaseModel, ConfigDict


class FunnelStageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    name: str
    count: int


class FunnelRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    total_applications: int
    stages: list[FunnelStageRead]
    submitted_rate: float | None
    response_rate: float | None
    interview_rate: float | None
    offer_rate: float | None


class ResumePerformanceRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    resume_id: str
    label: str
    total_applications: int
    submitted: int
    responses: int
    interviews: int
    offers: int
    avg_match_score: float | None
    response_rate: float | None


class TimelinePointRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    date: str
    discovered: int
    submitted: int


class AnalyticsSummaryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    total_applications: int
    total_submitted: int
    total_interviews: int
    total_offers: int
    avg_match_score: float | None
    avg_days_to_submit: float | None

File: frontend/src/types/analytics.ts
export interface FunnelStage {
  name: string;
  count: number;
}

export interface Funnel {
  total_applications: number;
  stages: FunnelStage[];
  submitted_rate: number | null;
  response_rate: number | null;
  interview_rate: number | null;
  offer_rate: number | null;
}

export interface ResumePerformance {
  resume_id: string;
  label: string;
  total_applications: number;
  submitted: number;
  responses: number;
  interviews: number;
  offers: number;
  avg_match_score: number | null;
  response_rate: number | null;
}

export interface TimelinePoint {
  date: string;
  discovered: number;
  submitted: number;
}

export interface AnalyticsSummary {
  total_applications: number;
  total_submitted: number;
  total_interviews: number;
  total_offers: number;
  avg_match_score: number | null;
  avg_days_to_submit: number | null;
}
